src/stream/stream/mod.rs

```diff
@@ -78,1089 +88,952 @@ cfg_if! {
     }
 }
 
-cfg_if! {
-    if #[cfg(feature = "docs")] {
-        use std::ops::{Deref, DerefMut};
+extension_trait! {
+    /// An asynchronous stream of values.
+    ///
+    /// This trait is a re-export of [`futures::stream::Stream`] and is an async version of
+    /// [`std::iter::Iterator`].
+    ///
+    /// The [provided methods] do not really exist in the trait itself, but they become
+    /// available when the prelude is imported:
+    ///
+    /// ```
+    /// # #[allow(unused_imports)]
+    /// use async_std::prelude::*;
+    /// ```
+    ///
+    /// [`std::iter::Iterator`]: https://doc.rust-lang.org/std/iter/trait.Iterator.html
+    /// [`futures::stream::Stream`]:
+    /// https://docs.rs/futures-preview/0.3.0-alpha.17/futures/stream/trait.Stream.html
+    /// [provided methods]: #provided-methods
+    pub trait Stream [StreamExt: futures_core::stream::Stream] {
+        /// The type of items yielded by this stream.
+        type Item;
+
+        #[doc = r#"
```

Comment:
We have to write docs in `Stream` in this style because the macro will otherwise exceed the recursion limit and dramatically slow compilation down. That is because each line starting with `///` is a token itself, but with `#[doc = "..."]` we can put the whole docs for a function into a single token tree, meaning the macro makes progress faster - instead of consuming docs line by line it consumes them in a single bite.